modules/database-migrations/lambda.tf

```diff
@@ -0,0 +1,93 @@
+resource "aws_iam_role" "lambda_assume_role" {
+  assume_role_policy = data.aws_iam_policy_document.lambda_assume_role_document.json
+  name               = "TDRDbMigrationLambdaRole${title(var.environment)}"
+}
+
+data "aws_iam_policy_document" "lambda_assume_role_document" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "sts:AssumeRole"
+    ]
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "lambda_migration_document" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:GetObject",
+      "s3:ListBucket"
+    ]
+    resources = [
+      "arn:aws:s3:::tdr-database-migrations/*",
+      "arn:aws:s3:::tdr-database-migrations"
+    ]
+  }
+  statement {
+    effect = "Allow"
+    actions = [
+      "ec2:DeleteNetworkInterface",
+      "ec2:DescribeInstances",
+      "ec2:CreateNetworkInterface",
+      "ec2:AttachNetworkInterface",
+      "ec2:DescribeNetworkInterfaces",
+      "autoscaling:CompleteLifecycleAction"
+    ]
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_policy" "lambda_migration_policy" {
+  name   = "TDRDbMigrationLambdaPolicy${title(var.environment)}"
+  policy = data.aws_iam_policy_document.lambda_migration_document.json
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_role_attach_migration_policy" {
+  policy_arn = aws_iam_policy.lambda_migration_policy.arn
+  role       = aws_iam_role.lambda_assume_role.name
+}
+
+resource "aws_lambda_function" "database_migration_function" {
+  function_name = "tdr-database-migrations-${var.environment}"
+  handler       = "migration.Main::runMigration"
+  role          = aws_iam_role.lambda_assume_role.arn
+  runtime       = "java8"
+  filename      = "${path.module}/temp.zip"
+  memory_size   = 128
+  timeout       = 60
```

Comment:
Should this be higher, just in case we have a DB migration that takes a while in production?

What do you think about having some kind of warning if we run a migration that runs close to the limit? e.g. the Jenkins migration job could be marked as unstable if it takes too long.

Reply:
I think this is ok for now and for a while yet. Anything running close to this at the moment is probably due to a failure which would be useful to know sooner.

Reply:
I've [created a card](https://github.com/orgs/nationalarchives/projects/20#card-32043820) for adding some kind of warning, and put it low down on the tech backlog.